In [2]:
mc_dropout_samples = 100
use_mc_dropout = True
dropout_probability = 0.4 if use_mc_dropout else 0.3

activation_function = "sigmoid"
skip_train_load_weights = False
target_column = "loan_status"
year_range = (2007, 2014)
num_epochs = 20
batch_size = 64
train_dev_test_split = (0.8, 0.1, 0.1)
embedding_dimension = 200
over_sampling_ratio = 0.5
weight_decay = 0.01
learning_rate = 0.001
random_state_for_split = 3213
transformer_config = (
    {
        "d_ff": 50,
        "max_seq_length": 75,
        "dropout": 0.3,
        "num_heads": 8, # TODO: should be 8, that is what the paper uses, embedding dimension must be divisible by num_heads
        "activation": "relu",
        "num_layers": 1,
    }
)
deep_feed_forward_hidden_units = (10, 10)

# TE runs

## No UQ no DFF dropout 64x64
- Final Devset AUC: 0.6948, Final Devset G-mean: 0.6426 | Test AUC: 0.7016, Test G-mean: 0.6486
- Final Devset AUC: 0.6856, Final Devset G-mean: 0.6374 | Test AUC: 0.7072, Test G-mean: 0.6507
- Final Devset AUC: 0.6999, Final Devset G-mean: 0.6385 | Test AUC: 0.6923, Test G-mean: 0.6249
- Final Devset AUC: 0.6797, Final Devset G-mean: 0.6273 | Test AUC: 0.7008, Test G-mean: 0.6437
- Final Devset AUC: 0.6984, Final Devset G-mean: 0.6059 | Test AUC: 0.6931, Test G-mean: 0.6228

## No UQ
<!-- - Final Devset AUC: 0.6748, Final Devset G-mean: 0.6274 | Test AUC: 0.7023, Test G-mean: 0.6423
- Final Devset AUC: 0.6968, Final Devset G-mean: 0.6495 | Test AUC: 0.7051, Test G-mean: 0.6519 -->
- Final Devset AUC: 0.6909, Final Devset G-mean: 0.6263 | Test AUC: 0.6888, Test G-mean: 0.6324
- Final Devset AUC: 0.6968, Final Devset G-mean: 0.6378 | Test AUC: 0.6850, Test G-mean: 0.6283
- Final Devset AUC: 0.6894, Final Devset G-mean: 0.6226 | Test AUC: 0.7033, Test G-mean: 0.6419
- Final Devset AUC: 0.6807, Final Devset G-mean: 0.6323 | Test AUC: 0.7030, Test G-mean: 0.6582
- Final Devset AUC: 0.7009, Final Devset G-mean: 0.6543 | Test AUC: 0.6944, Test G-mean: 0.6396


In [3]:
import numpy as np

auc = [0.6909,
       0.6968,
       0.6894,
       0.6807,
       0.7009]
gmean = [0.6263,
         0.6378,
         0.6226,
         0.6323,
         0.6543]
test_auc = [0.6888,
            0.6850,
            0.7033,
            0.7030,
            0.6944]
test_gmean = [0.6324,
              0.6283,
              0.6419,
              0.6582,
              0.6396]


def mean_std(data):
    mean = np.mean(data)
    std = np.std(data)
    return mean, std

def calculate_statistics(auc, gmean, test_auc, test_gmean):
       mean_auc, std_auc = mean_std(auc)
       mean_gmean, std_gmean = mean_std(gmean)
       mean_test_auc, std_test_auc = mean_std(test_auc)
       mean_test_gmean, std_test_gmean = mean_std(test_gmean)

       print(f"Mean AUC: {mean_auc}, Std AUC: {std_auc}")
       print(f"Mean G-mean: {mean_gmean}, Std G-mean: {std_gmean}")
       print(f"Mean Test AUC: {mean_test_auc}, Std Test AUC: {std_test_auc}")
       print(f"Mean Test G-mean: {mean_test_gmean}, Std Test G-mean: {std_test_gmean}")

calculate_statistics(auc, gmean, test_auc, test_gmean)

Mean AUC: 0.6917399999999999, Std AUC: 0.0068930689826810785
Mean G-mean: 0.63466, Std G-mean: 0.011103981267995723
Mean Test AUC: 0.6949, Std Test AUC: 0.007370753014448377
Mean Test G-mean: 0.64008, Std Test G-mean: 0.01029201632334502


## Aleatoric UQ
1.
DEBUG:__main__:Aleatoric mean, min & max STD [0.7957, 0.6654, 0.8312]
INFO:__main__:Final Devset AUC: 0.7091, Final Devset G-mean: 0.6512, Aleatoric error correlation: 0.6597
INFO:__main__:Final Devset AUC with high confidence: 0.7647, Final Devset G-mean with high confidence: 0.6747

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.7959, 0.6584, 0.8312]
INFO:__main__:Test AUC: 0.6827, Test G-mean: 0.6322, Aleatoric error correlation: 0.6474
INFO:__main__:Test AUC with high confidence: 0.7408, Test G-mean with high confidence: 0.6646

2.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8099, 0.7440, 0.8385]
INFO:__main__:Final Devset AUC: 0.6832, Final Devset G-mean: 0.6338, Aleatoric error correlation: 0.5255
INFO:__main__:Final Devset AUC with high confidence: 0.7434, Final Devset G-mean with high confidence: 0.6854

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.8098, 0.7393, 0.8383]
INFO:__main__:Test AUC: 0.7001, Test G-mean: 0.6514, Aleatoric error correlation: 0.5273
INFO:__main__:Test AUC with high confidence: 0.7565, Test G-mean with high confidence: 0.7121

3.
DEBUG:__main__:Aleatoric mean, min & max STD [0.7793, 0.6375, 0.8230]
INFO:__main__:Final Devset AUC: 0.6992, Final Devset G-mean: 0.6445, Aleatoric error correlation: 0.5654
INFO:__main__:Final Devset AUC with high confidence: 0.7546, Final Devset G-mean with high confidence: 0.6961

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.7791, 0.6484, 0.8210]
INFO:__main__:Test AUC: 0.6961, Test G-mean: 0.6427, Aleatoric error correlation: 0.5725
INFO:__main__:Test AUC with high confidence: 0.7549, Test G-mean with high confidence: 0.6938

4.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8014, 0.7203, 0.8331]
INFO:__main__:Final Devset AUC: 0.6929, Final Devset G-mean: 0.6377, Aleatoric error correlation: 0.4769, 
INFO:__main__:Final Devset AUC with high confidence: 0.7545, Final Devset G-mean with high confidence: 0.7046

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.8016, 0.6857, 0.8325]
INFO:__main__:Test AUC: 0.6969, Test G-mean: 0.6460, Aleatoric error correlation: 0.4871, 
INFO:__main__:Test AUC with high confidence: 0.7545, Test G-mean with high confidence: 0.7114

5.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8070, 0.6998, 0.8378]
INFO:__main__:Final Devset AUC: 0.7020, Final Devset G-mean: 0.6519, Aleatoric error correlation: 0.5107, 
INFO:__main__:Final Devset AUC with high confidence: 0.7633, Final Devset G-mean with high confidence: 0.7072

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.8075, 0.6843, 0.8364]
INFO:__main__:Test AUC: 0.6872, Test G-mean: 0.6414, Aleatoric error correlation: 0.5191, 
INFO:__main__:Test AUC with high confidence: 0.7449, Test G-mean with high confidence: 0.6983

In [12]:

dev_auc_au = [
0.7091,
 0.6772,
 0.6992,
 0.6929,
 0.7020
]
dev_gmean_au = [
0.6512,
0.6195,
0.6445,
0.6377,
0.6519]

test_au = {
    "test_auc_au":          [0.6827, 0.7001, 0.6961, 0.6969, 0.6872],
    "test_gmean_au":        [0.6322, 0.6514, 0.6427, 0.6460, 0.6414],
    "test_corr_au":         [0.6474, 0.5273, 0.5725, 0.4871, 0.5191],
    "test_auc_high_conf":   [0.7408, 0.7565, 0.7549, 0.7545, 0.7449],
    "test_gmean_high_conf": [0.6646, 0.7121, 0.6938, 0.7114, 0.6983],
    "test_au_mean":         [0.7959, 0.8098, 0.7791, 0.8016, 0.8075]
}


for key, value in test_au.items():
    print(f"{key}: {mean_std(value)}")

test_auc_au: (0.6926, 0.006544616107916482)
test_gmean_au: (0.64274, 0.006300984050130585)
test_corr_au: (0.5506800000000001, 0.05552874570886686)
test_auc_high_conf: (0.7503200000000001, 0.006271331597037407)
test_gmean_high_conf: (0.6960400000000001, 0.017276874717378727)
test_au_mean: (0.79878, 0.01096383144708088)


## Aleatoric & Epistemic UQ
1.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8094, 0.7383, 0.8315]
DEBUG:__main__:Epistemic mean, min & max STD [0.0626, 0.0124, 0.1566]
INFO:__main__:Final Devset AUC: 0.6960, Final Devset G-mean: 0.6476, Aleatoric error correlation: 0.3168, Epistemic error correlation: -0.2593
INFO:__main__:Final Devset AUC with high confidence: 0.7483, Final Devset G-mean with high confidence: 0.7065

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.8091, 0.7292, 0.8321]
DEBUG:__main__:Epistemic mean, min & max STD [0.0631, 0.0129, 0.1727]
INFO:__main__:Test AUC: 0.6897, Test G-mean: 0.6404, Aleatoric error correlation: 0.2943, Epistemic error correlation: -0.2340       
INFO:__main__:Test AUC with high confidence: 0.7481, Test G-mean with high confidence: 0.7021

2.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8059, 0.7164, 0.8301]
DEBUG:__main__:Epistemic mean, min & max STD [0.0624, 0.0175, 0.1480]
INFO:__main__:Final Devset AUC: 0.6983, Final Devset G-mean: 0.6425, Aleatoric error correlation: 0.5493, Epistemic error correlation: -0.4415
INFO:__main__:Final Devset AUC with high confidence: 0.7522, Final Devset G-mean with high confidence: 0.7108

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.8058, 0.7004, 0.8308]
DEBUG:__main__:Epistemic mean, min & max STD [0.0627, 0.0171, 0.1702]
INFO:__main__:Test AUC: 0.6811, Test G-mean: 0.6285, Aleatoric error correlation: 0.5214, Epistemic error correlation: -0.4073       
INFO:__main__:Test AUC with high confidence: 0.7341, Test G-mean with high confidence: 0.6899

3.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8084, 0.7375, 0.8333]
DEBUG:__main__:Epistemic mean, min & max STD [0.0621, 0.0148, 0.1860]
INFO:__main__:Test AUC: 0.6940, Test G-mean: 0.6363, Aleatoric error correlation: 0.5590, Epistemic error correlation: -0.4892       
INFO:__main__:Test AUC with high confidence: 0.7518, Test G-mean with high confidence: 0.6900
4.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8176, 0.7611, 0.8349]
DEBUG:__main__:Epistemic mean, min & max STD [0.0598, 0.0110, 0.1488]
INFO:__main__:Test AUC: 0.6931, Test G-mean: 0.6401, Aleatoric error correlation: 0.5069, Epistemic error correlation: -0.5637       
INFO:__main__:Test AUC with high confidence: 0.7503, Test G-mean with high confidence: 0.7180
5.
DEBUG:__main__:Aleatoric mean, min & max STD [0.8162, 0.7577, 0.8345]
DEBUG:__main__:Epistemic mean, min & max STD [0.0574, 0.0116, 0.1442]
INFO:__main__:Test AUC: 0.6958, Test G-mean: 0.6452, Aleatoric error correlation: 0.6890, Epistemic error correlation: -0.6603       
INFO:__main__:Test AUC with high confidence: 0.7435, Test G-mean with high confidence: 0.6557

In [9]:
combined_uq = {
    "test_auc":             [0.6897, 0.6811, 0.6940, 0.6931, 0.6958],
    "test_gmean":           [0.6404, 0.6285, 0.6363, 0.6401, 0.6452],
    "test_aucorr":          [0.2943, 0.5214, 0.5590, 0.5069, 0.6890],
    "test_eucorr":          [-0.2340, 0.4073, -0.4892, -0.5637, -0.6603],
    "test_auc_high_conf":   [0.7481, 0.7341, 0.7518, 0.7503, 0.7435],
    "test_gmean_high_conf": [0.7021, 0.6899, 0.6900, 0.7180, 0.6557],
    "epistemic_mean":       [0.0631, 0.0627, 0.0621, 0.0598, 0.0574],
    "aleatoric_mean":       [0.8091, 0.8058, 0.8084, 0.8176, 0.8162]
}

for key, value in combined_uq.items():
    mean, std = mean_std(value)
    print(f"{key}: {mean} ({std})")

total_uq_mean = [x + y for x, y in zip(combined_uq["epistemic_mean"], combined_uq["aleatoric_mean"])]
mean, std = mean_std(total_uq_mean)
print(f"Total UQ: {mean} ({std})")


test_auc: 0.6907400000000001 (0.005211755942098573)
test_gmean: 0.6380999999999999 (0.005569560126257744)
test_aucorr: 0.51412 (0.127297437523306)
test_eucorr: -0.30798000000000003 (0.38456762942296635)
test_auc_high_conf: 0.74556 (0.006378589185705571)
test_gmean_high_conf: 0.69114 (0.020495911787476062)
epistemic_mean: 0.06102 (0.002140467238712148)
aleatoric_mean: 0.81142 (0.004628779536767775)
Total UQ: 0.8724399999999999 (0.003009717594725456)


## E UQ

1.
DEBUG:__main__:Epistemic mean, min & max STD [0.0593, 0.0217, 0.1482]
INFO:__main__:Final Devset AUC: 0.7002, Final Devset G-mean: 0.6428, Aleatoric error correlation: nan, Epistemic error correlation: -0.0599
INFO:__main__:Final Devset AUC with high confidence: 0.6130, Final Devset G-mean with high confidence: 0.5163

2.
DEBUG:__main__:Epistemic mean, min & max STD [0.0639, 0.0130, 0.1452]
INFO:__main__:Final Devset AUC: 0.7000, Final Devset G-mean: 0.6523, Aleatoric error correlation: nan, Epistemic error correlation: -0.6396
INFO:__main__:Final Devset AUC with high confidence: 0.6123, Final Devset G-mean with high confidence: 0.5598

3.
DEBUG:__main__:Epistemic mean, min & max STD [0.0587, 0.0152, 0.1318]
INFO:__main__:Final Devset AUC: 0.7005, Final Devset G-mean: 0.6548, Aleatoric error correlation: nan, Epistemic error correlation: -0.4098
INFO:__main__:Final Devset AUC with high confidence: 0.6037, Final Devset G-mean with high confidence: 0.5885

4.
DEBUG:__main__:Epistemic mean, min & max STD [0.0625, 0.0101, 0.1564]
INFO:__main__:Final Devset AUC: 0.6894, Final Devset G-mean: 0.6409, Aleatoric error correlation: nan, Epistemic error correlation: -0.6014
INFO:__main__:Final Devset AUC with high confidence: 0.5955, Final Devset G-mean with high confidence: 0.5472

5.
DEBUG:__main__:Epistemic mean, min & max STD [0.0687, 0.0129, 0.1842]
INFO:__main__:Final Devset AUC: 0.6905, Final Devset G-mean: 0.6395, Aleatoric error correlation: nan, Epistemic error correlation: -0.5305
INFO:__main__:Final Devset AUC with high confidence: 0.5998, Final Devset G-mean with high confidence: 0.5725

## debug^

1.
DEBUG:__main__:Epistemic mean, min & max STD [0.0599, 0.0201, 0.1533]
INFO:__main__:Test AUC: 0.6869, Test G-mean: 0.6220, Aleatoric error correlation: nan, Epistemic error correlation: -0.0721
INFO:__main__:Test AUC with high confidence: 0.5975, Test G-mean with high confidence: 0.4982

2.
DEBUG:__main__:Epistemic mean, min & max STD [0.0646, 0.0135, 0.1543]
INFO:__main__:Test AUC: 0.6898, Test G-mean: 0.6322, Aleatoric error correlation: nan, Epistemic error correlation: -0.6571
INFO:__main__:Test AUC with high confidence: 0.5885, Test G-mean with high confidence: 0.5366


3.
DEBUG:__main__:Epistemic mean, min & max STD [0.0598, 0.0156, 0.1558]
INFO:__main__:Test AUC: 0.6865, Test G-mean: 0.6310, Aleatoric error correlation: nan, Epistemic error correlation: -0.4277
INFO:__main__:Test AUC with high confidence: 0.5756, Test G-mean with high confidence: 0.5550

4.
DEBUG:__main__:Epistemic mean, min & max STD [0.0627, 0.0095, 0.1518]
INFO:__main__:Test AUC: 0.7037, Test G-mean: 0.6494, Aleatoric error correlation: nan, Epistemic error correlation: -0.6096
INFO:__main__:Test AUC with high confidence: 0.6033, Test G-mean with high confidence: 0.5512

5.
DEBUG:__main__:Epistemic mean, min & max STD [0.0690, 0.0134, 0.1723]
INFO:__main__:Test AUC: 0.7021, Test G-mean: 0.6451, Aleatoric error correlation: nan, Epistemic error correlation: -0.5368
INFO:__main__:Test AUC with high confidence: 0.6126, Test G-mean with high confidence: 0.5735

In [6]:
data = {
    "test_auc": [0.6869, 0.6898, 0.6865, 0.7037, 0.7021],
    "test_gmean": [0.6220, 0.6322, 0.6310, 0.6494, 0.6451],
    "epistemic_corr": [-0.0721, -0.6571, -0.4277, -0.6096, -0.5368],
    "test_auc_high": [0.5975, 0.5885, 0.5756, 0.6033, 0.6126],
    "test_gmean_high": [0.4982, 0.5366, 0.5550, 0.5512, 0.5735],
    "epistemic_mean": [0.0599, 0.0646, 0.0598, 0.0627, 0.0690]
}

for key, value in data.items():
    mean, std = mean_std(value)
    print(f"{key}: {mean} ({std})")

test_auc: 0.6938000000000001 (0.007533923280734943)
test_gmean: 0.6359400000000001 (0.009977895569708071)
epistemic_corr: -0.46066 (0.20910969944026986)
test_auc_high: 0.5955 (0.012666964908769583)
test_gmean_high: 0.5429 (0.02525644472209026)
epistemic_mean: 0.0632 (0.0034146742157927764)


## Combined UQ with no sigmoid
Dev.
DEBUG:__main__:Aleatoric mean, min & max STD [0.4820, 0.4134, 0.4967]
DEBUG:__main__:Epistemic mean, min & max STD [0.0740, 0.0246, 0.2756]
INFO:__main__:Final Devset AUC: 0.6945, Final Devset G-mean: 0.6425, Aleatoric error correlation: 0.5097, Epistemic error correlation: -0.3795
INFO:__main__:Final Devset AUC with high confidence: 0.7541, Final Devset G-mean with high confidence: 0.7198
DEBUG:__main__:Aleatoric mean, min & max STD [0.4873, 0.4117, 0.5059]
DEBUG:__main__:Epistemic mean, min & max STD [0.0707, 0.0221, 0.2321]
INFO:__main__:Final Devset AUC: 0.6946, Final Devset G-mean: 0.6425, Aleatoric error correlation: 0.5381, Epistemic error correlation: -0.4726
INFO:__main__:Final Devset AUC with high confidence: 0.7523, Final Devset G-mean with high confidence: 0.6982
DEBUG:__main__:Aleatoric mean, min & max STD [0.4851, 0.4071, 0.5037]
DEBUG:__main__:Epistemic mean, min & max STD [0.0670, 0.0187, 0.2225]
INFO:__main__:Final Devset AUC: 0.6951, Final Devset G-mean: 0.6395, Aleatoric error correlation: 0.6422, Epistemic error correlation: -0.5333
INFO:__main__:Final Devset AUC with high confidence: 0.7421, Final Devset G-mean with high confidence: 0.6654

INFO:__main__:Evaluating on test set
DEBUG:__main__:Aleatoric mean, min & max STD [0.4817, 0.4179, 0.4963]
DEBUG:__main__:Epistemic mean, min & max STD [0.0747, 0.0237, 0.2804]
INFO:__main__:Test AUC: 0.6795, Test G-mean: 0.6265, Aleatoric error correlation: 0.5026, Epistemic error correlation: -0.3708       
INFO:__main__:Test AUC with high confidence: 0.7366, Test G-mean with high confidence: 0.7043

DEBUG:__main__:Aleatoric mean, min & max STD [0.4870, 0.4217, 0.5060]
DEBUG:__main__:Epistemic mean, min & max STD [0.0713, 0.0209, 0.2145]
INFO:__main__:Test AUC: 0.6802, Test G-mean: 0.6302, Aleatoric error correlation: 0.5278, Epistemic error correlation: -0.4548       
INFO:__main__:Test AUC with high confidence: 0.7359, Test G-mean with high confidence: 0.6959

DEBUG:__main__:Aleatoric mean, min & max STD [0.4849, 0.4205, 0.5030]
DEBUG:__main__:Epistemic mean, min & max STD [0.0676, 0.0178, 0.1998]
INFO:__main__:Test AUC: 0.6792, Test G-mean: 0.6288, Aleatoric error correlation: 0.6408, Epistemic error correlation: -0.5217       
INFO:__main__:Test AUC with high confidence: 0.7198, Test G-mean with high confidence: 0.6668

INFO:__main__:Evaluating on test set

DEBUG:__main__:AU mean, min & max STD per class: [0.4852, 0.4116, 0.5010], [0.4938, 0.4489, 0.5014]
DEBUG:__main__:EU mean, min & max STD per clas :  [0.0791, 0.0239, 0.3067], [0.0504, 0.0248, 0.2021]
DEBUG:__main__:AU error correlation per class: 0.9725, -0.9617
DEBUG:__main__:EU error correlation per class: -0.9682, 0.9645
DEBUG:__main__:Aleatoric mean, min & max STD [0.4864, 0.4116, 0.5014]
DEBUG:__main__:Epistemic mean, min & max STD [0.0750, 0.0239, 0.3067]
INFO:__main__:Test AUC: 0.7010, Test G-mean: 0.6517, Aleatoric error correlation: 0.8096, Epistemic error correlation: -0.7977       
INFO:__main__:Test AUC with high confidence: 0.6458, Test G-mean with high confidence: 0.0000

DEBUG:__main__:AU mean, min & max STD per class: [0.5973, 0.4953, 0.8191], [0.6127, 0.5349, 0.7915]
DEBUG:__main__:EU mean, min & max STD per clas :  [0.1545, 0.0934, 1.6714], [0.1420, 0.0955, 1.0152]
DEBUG:__main__:AU error correlation per class: 0.6452, -0.4811
DEBUG:__main__:EU error correlation per class: -0.6013, 0.6490
DEBUG:__main__:Aleatoric mean, min & max STD [0.5995, 0.4953, 0.8191]
DEBUG:__main__:Epistemic mean, min & max STD [0.1527, 0.0934, 1.6714]
INFO:__main__:Test AUC: 0.6773, Test G-mean: 0.1171, Aleatoric error correlation: 0.1030, Epistemic error correlation: -0.1504       
INFO:__main__:Test AUC with high confidence: 0.6609, Test G-mean with high confidence: 0.1206

3. HTML file for this run
DEBUG:__main__:AU mean, min & max STD per class: [0.4833, 0.3841, 0.5007], [0.4878, 0.4337, 0.5001]
DEBUG:__main__:EU mean, min & max STD per clas :  [0.0654, 0.0232, 0.2483], [0.0605, 0.0237, 0.1337]
DEBUG:__main__:AU error correlation per class: 0.7123, -0.1281
DEBUG:__main__:EU error correlation per class: -0.5634, -0.1581
DEBUG:__main__:Aleatoric mean, min & max STD [0.4839, 0.3841, 0.5007]
DEBUG:__main__:Epistemic mean, min & max STD [0.0647, 0.0232, 0.2483]
INFO:__main__:Test AUC: 0.7001, Test G-mean: 0.6419, Aleatoric error correlation: 0.6198, Epistemic error correlation: -0.5078       
INFO:__main__:Test AUC with high confidence: 0.7589, Test G-mean with high confidence: 0.7262

4. HTML file for this run
DEBUG:__main__:AU mean, min & max STD per class: [0.4834, 0.4238, 0.4994], [0.4858, 0.4353, 0.4990]
DEBUG:__main__:EU mean, min & max STD per clas :  [0.0753, 0.0156, 0.2235], [0.0698, 0.0163, 0.1962]
DEBUG:__main__:AU error correlation per class: 0.6229, -0.0119
DEBUG:__main__:EU error correlation per class: -0.5467, -0.0889
DEBUG:__main__:Aleatoric mean, min & max STD [0.4838, 0.4238, 0.4994]
DEBUG:__main__:Epistemic mean, min & max STD [0.0745, 0.0156, 0.2235]
INFO:__main__:Test AUC: 0.6811, Test G-mean: 0.6388, Aleatoric error correlation: 0.5548, Epistemic error correlation: -0.4957
INFO:__main__:Test AUC with high confidence: 0.7353, Test G-mean with high confidence: 0.6859

5. HTML file for this run
DEBUG:__main__:AU mean, min & max STD per class: [0.5342, 0.4734, 0.5772], [0.5414, 0.4821, 0.5741]
DEBUG:__main__:EU mean, min & max STD per clas :  [0.0735, 0.0287, 0.3336], [0.0598, 0.0315, 0.1719]
DEBUG:__main__:AU error correlation per class: 0.8188, -0.5922
DEBUG:__main__:EU error correlation per class: -0.7963, 0.5379
DEBUG:__main__:Aleatoric mean, min & max STD [0.5353, 0.4734, 0.5772]
DEBUG:__main__:Epistemic mean, min & max STD [0.0716, 0.0287, 0.3336]
INFO:__main__:Test AUC: 0.7022, Test G-mean: 0.6331, Aleatoric error correlation: 0.6203, Epistemic error correlation: -0.6233
INFO:__main__:Test AUC with high confidence: 0.7095, Test G-mean with high confidence: 0.6201